api/v1alpha2/istio_structs.go

```diff
@@ -103,3 +106,76 @@ type ResourceClaims struct {
 	// +kubebuilder:validation:Pattern=`^[0-9]+(((\.[0-9]+)?(E|P|T|G|M|k|Ei|Pi|Ti|Gi|Mi|Ki|m)?)|(e[0-9]+))$`
 	Memory *string `json:"memory,omitempty"`
 }
+
+type Authorizer struct {
+	// A unique name identifying the extension authorization provider.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Specifies the service that implements the Envoy ext_authz HTTP authorization service.
+	// The format is "[<Namespace>/]<Hostname>".
+	// The specification of "<Namespace>"
+	// is required only when it is insufficient to unambiguously resolve a service in the service registry.
+	// The "<Hostname>" is a fully qualified host name of a service defined by the Kubernetes service or ServiceEntry.
+	// The recommended format is "[<Namespace>/]<Hostname>"
+	// Example: "my-ext-authz.foo.svc.cluster.local" or "bar/my-ext-authz".
+	// +kubebuilder:validation:Required
+	Service string `json:"service"`
```

Comment:
We don't support external services outside of the mesh, because we'd need to set `location: MESH_EXTERNAL` in the ServiceEntry. Do we need to update the ADR and/or have it clearly documented anywhere?